azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java

```diff
@@ -359,6 +372,15 @@ public void fillExecutableFromMapObject(
       this.slaOptionStr = slaBuilder.toString();
     }
 
+    if (flowObj.containsKey(VERSIONSET_JSON_PARAM) && flowObj.containsKey(VERSIONSET_MD5HEX_PARAM) && flowObj.containsKey(VERSIONSET_ID_PARAM)) {
```

Comment:
Too long, break into multiple lines.
I believe all 3 properties are added as a block so instead of checking for all of them, check for one and log.warn for others if they are missing.
If someone changes the code in the future, such coding errors will be caught that way.